lib/calendar_assistant/calendar_assistant.rb

```diff
@@ -56,72 +56,42 @@ def in_tz &block
     end
-
-    event_repository(calendar_ids.first).find(time_range,  predicates: @event_predicates.merge({needs_action?: true}))
+    event_repository(calendar_ids.first, type: :lint).find(time_range,  predicates: @event_predicates)
   end
 
   def find_events time_range
-    calendar_ids = config.attendees
+    calendar_ids = config.calendar_ids
     if calendar_ids.length > 1
       raise BaseException, "CalendarAssistant#find_events only supports one person (for now)"
     end
     event_repository(calendar_ids.first).find(time_range, predicates: @event_predicates)
   end
 
   def availability time_range
-    calendar_ids = config.attendees
+    calendar_ids = config.calendar_ids
     ers = calendar_ids.map do |calendar_id|
       event_repository calendar_id
     end
     Scheduler.new(self, ers).available_blocks(time_range, predicates: @event_predicates)
   end
 
   def find_location_events time_range
-    event_set = event_repository.find(time_range, predicates: @event_predicates)
-    event_set.new event_set.events.select { |e| e.location_event? }
+    event_repository(type: :location).find(time_range, predicates: @event_predicates)
   end
 
-  def create_location_event time_range, location
-    # find pre-existing events that overlap
-    existing_event_set = find_location_events time_range
-
-    # augment event end date appropriately
-    range = CalendarAssistant.date_range_cast time_range
-
-    deleted_events = []
-    modified_events = []
-
-    event = event_repository.create(
-      transparency: CalendarAssistant::Event::Transparency::TRANSPARENT,
-      start: range.first, end: range.last,
-      summary: "#{Event.location_event_prefix(@config)}#{location}"
-    )
-
-    existing_event_set.events.each do |existing_event|
-      if existing_event.start_date >= event.start_date && existing_event.end_date <= event.end_date
-        event_repository.delete existing_event
-        deleted_events << existing_event
-      elsif existing_event.start_date <= event.end_date && existing_event.end_date > event.end_date
-        event_repository.update existing_event, start: range.last
-        modified_events << existing_event
-      elsif existing_event.start_date < event.start_date && existing_event.end_date >= event.start_date
-        event_repository.update existing_event, end: range.first
-        modified_events << existing_event
-      end
-    end
-
-    response = {created: [event]}
-    response[:deleted] = deleted_events unless deleted_events.empty?
-    response[:modified] = modified_events unless modified_events.empty?
-
-    existing_event_set.new response
+  def create_location_events time_range, location
+    EventSet.new(event_repository, (Array(config.calendar_ids) | [Config::DEFAULT_CALENDAR_ID]).each_with_object({}) do |calendar_id, hsh|
+      hsh[calendar_id] = event_repository(calendar_id, type: :location).create(time_range, location, predicates: @event_predicates)
+    end)
```

Comment:
it took me like two minutes to parse what you're doing here. would you be opposed to changing this to be

```
events = (Array(config.calendar_ids) | [Config::DEFAULT_CALENDAR_ID]).each_with_object({}) do |calendar_id, hsh|
      hsh[calendar_id] = event_repository(calendar_id, type: :location).create(time_range, location, predicates: @event_predicates)
    end
EventSet.new(event_repository, events)
```
?

Reply:
:excited:

Reply:
This is fixed, check out the new code!